java/src/jmri/SectionManager.java

```diff
@@ -237,6 +239,91 @@ public void initializeBlockingSensors() {
             }
         }
     }
+        
+    /**
+     * Generate Block Sections in stubs/sidings. Called after generating signal logic.
+     */
+    
+    
+    public void generateBlockSections() {
+        //find blocks with no paths through i.e. stub (siding)
+        LayoutBlockManager LayoutBlockManager = jmri.InstanceManager.getDefault(LayoutBlockManager.class);
+        //print "Layout Block"
+        for (LayoutBlock layoutBlock : LayoutBlockManager.getNamedBeanSet()){
+            if (layoutBlock.getNumberOfThroughPaths() == 0){
+                if (!blockSectionExists(layoutBlock)){
+                    //create block section"
+                    createBlockSection(layoutBlock);
+                }
+            }
+        }
+    }
+                               
+    /**
+     * Check if Block Section already exists
+     * @param layoutBlock
+     * @return true or false
+     */
+    private boolean blockSectionExists(LayoutBlock layoutBlock){
+        
+        for (Section section : getNamedBeanSet()){
+            if (section.getNumBlocks() == 1 
+                    && section.getSectionType() != Section.SIGNALMASTLOGIC 
+                    && section.getEntryBlock() == layoutBlock.getBlock()){
+                return true;
+            }
+        }
+        return false;
+    } 
+      
+    private void createBlockSection(LayoutBlock layoutBlock){
+        Section section = createNewSection(layoutBlock.getUserName()); 
+        section.addBlock(layoutBlock.getBlock());
+        java.util.ArrayList<jmri.EntryPoint> entryPointList = new java.util.ArrayList<jmri.EntryPoint>();
```

Comment:
ArrayList is in the imports section, java.util not needed.   The `jmri.EntryPoint` after the new is not needed: `new ArrayList<>().